edge/pkg/metamanager/apiserver-lite/kubernetes/storage/storage.go

```diff
@@ -0,0 +1,68 @@
+package storage
+
+import (
+	"github.com/kubeedge/kubeedge/edge/pkg/metamanager/apiserver-lite/kubernetes/storage/cacher"
+	"github.com/kubeedge/kubeedge/pkg/apiserverlite"
+	"github.com/kubeedge/kubeedge/pkg/apiserverlite/util"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	genericregistry "k8s.io/apiserver/pkg/registry/generic/registry"
+	"k8s.io/apiserver/pkg/storage"
+	"k8s.io/klog/v2"
+)
+
+// REST implements a RESTStorage for all resource against imitator.
+type REST struct {
+	*genericregistry.Store
+}
+
+// NewREST returns a RESTStorage object that will work against all resources
+func NewREST() (*REST, error) {
+	store := &genericregistry.Store{
+		NewFunc:                  func() runtime.Object { return &unstructured.Unstructured{} },
+		NewListFunc:              func() runtime.Object { return &unstructured.UnstructuredList{} },
+		DefaultQualifiedResource: schema.GroupResource{},
+
+		KeyFunc: apiserverlite.KeyFuncReq,
+		KeyRootFunc: apiserverlite.KeyRootFunc,
+
+		CreateStrategy: nil,
+		UpdateStrategy: nil,
+		DeleteStrategy: nil,
+
+		TableConvertor: nil,
+		StorageVersioner : nil,
+		Storage: genericregistry.DryRunnableStorage{},
+	}
+	store.PredicateFunc = func(label labels.Selector, field fields.Selector) storage.SelectionPredicate {
+		return storage.SelectionPredicate{
+			Label:    label,
+			Field:    field,
+			GetAttrs: util.UnstructuredAttr,
+		}
+	}
+	store.Decorator = func(obj runtime.Object) error {
+		unstrList,ok :=obj.(*unstructured.UnstructuredList)
+		if ok {
+			var gvk schema.GroupVersionKind
+			if len(unstrList.Items) == 0{
+				klog.Errorf("failed to decorate an empty list")
+				return nil
+			}
+			gvk = unstrList.Items[0].GroupVersionKind()
+			gvk.Kind = gvk.Kind + "List"
+			unstrList.GetObjectKind().SetGroupVersionKind(gvk)
+		}
+		return nil
+	}
+	Storage,err:= cacher.NewCacher()
```

Comment:
What if we don't use cache here?

Reply:
@fisherxu cacher handles all the watch req(rev != 0). if remove it，backend storage (at SQLite/store.go) need to support handling such watch req，it has not been competed now，and even maybe a same realization like cacher .

Reply:
@fisherxu or we could cache history message(history resource) at sqlite. It is in TODO，may next pr?